### ISSUE TITLE
fit_generator log shows 1 sample each time 

### ISSUE THREAD

**Issue by HikaruL** [keras-team/keras#7880]
I have a generator which read from the txt file and generate samples indefinitely. Here is my code

```python
def myGenerator(file, batch_size, num_class, context_num):
    f = open(file)
    while 1:
        batch_x_left = []
        batch_x_middle = []
        batch_x_right = []
        batch_y = []
        for line in f:
            # create Numpy arrays of input data
            # and labels, from each line in the file
            if line.strip():
                array = line.strip().split(',')
                x = array[:-1]
                batch_x_left.append(x[:context_num * 262])
                batch_x_right.append(x[context_num * 262:-1])
                batch_x_middle.append(x[-2])
                batch_y.append(x[-1])
                if len(batch_x_left) >= batch_size:
                    break
        if len(batch_x_left) < batch_size:
            f = open(file)
            for line in f:
                # create Numpy arrays of input data
                # and labels, from each line in the file
                if line.strip():
                    array = line.strip().split(',')
                    x = array[:-1]
                    batch_x_left.append(x[:context_num * 262])
                    batch_x_right.append(x[context_num * 262:-1])
                    batch_x_middle.append(x[-2])
                    batch_y.append(x[-1])
                    if len(batch_x_left) >= batch_size:
                        break
        batch_x_left = np.array(batch_x_left)
        batch_x_right = np.array(batch_x_right)
        batch_x_middle = np.array(batch_x_middle)
        # print('length of real batch: ', len(x))
        batch_x_left = batch_x_left.astype('float32')
        batch_x_right = batch_x_right.astype('float32')
        batch_x_middle = batch_x_middle.astype('float32')
        y = np.array(batch_y)
        y = y.astype('float32')
        batch_x_middle = keras.utils.to_categorical(batch_x_middle, 62)
        y = keras.utils.to_categorical(y, num_class)
        batch_x_left = batch_x_left.reshape(batch_size, context_num, 262)
        batch_x_right = batch_x_right.reshape(batch_size, context_num, 262)
        yield [batch_x_left, batch_x_middle, batch_x_right], y
    f.close()
num_class = 34116
context_num = 6

model.fit_generator(myGenerator('lstm_training_data.txt', 1024, num_class, context_num), samples_per_epoch=14812565, nb_epoch=20, verbose=1, class_weight=None, nb_worker=1)
```
But the log is like this, the sample size on the left increase 1 each time. I'm not sure which part of my generator is wrong.

Epoch 1/20
2017-09-12 12:08:53.811817: I tensorflow/core/common_runtime/gpu/pool_allocator.cc:247] PoolAllocator: After 2432 get requests, put_count=2141 evicted_count=
2017-09-12 12:08:53.811855: I tensorflow/core/common_runtime/gpu/pool_allocator.cc:259] Raising pool_size_limit_ from 100 to 110
       9/14812565 [..............................] - ETA: 32661942s - loss: 9.5146 - acc: 0.27432017-09-12 12:09:11.062317: I tensorflow/core/common_runtime/9 evicted_count=1000 eviction_rate=0.438789 and unsatisfied allocation rate=0.474916
      19/14812565 [..............................] - ETA: 30636020s - loss: 7.9542 - acc: 0.31332017-09-12 12:09:30.449876: I tensorflow/core/common_runtime/5 evicted_count=1000 eviction_rate=0.362976 and unsatisfied allocation rate=0.302632
